### PR TITLE
Added cumem

### PR DESCRIPTION
**Pull request by BinhaoQin**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

Revision of merge request #1415

### CODE REVIEW

File: packages/c/cumem/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package("cumem")
+    set_kind("library", { headeronly = true })
+    set_description("CUDA Memory Management Wrapper with Type Safety")
+    set_homepage("https://github.com/BinhaoQin/cuMem")
```
Comment (waruqi) [inline packages/c/cumem/xmake.lua:4]: add an empty line

File: packages/c/cumem/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    add_urls("https://github.com/BinhaoQin/cuMem/archive/refs/tags/$(version).tar.gz")
+    add_versions("1.0.0", "1c7956c840f7aa3940756c8d09a2ae44ce53681ae01f3378cf4501d81bf0638c")
```
Comment (waruqi) [inline packages/c/cumem/xmake.lua:6]: and here

File: packages/c/cumem/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    add_deps("cuda", { system = true })
+
+    on_install(function(package)
```
Comment (waruqi) [inline packages/c/cumem/xmake.lua:9]: you need limit the supported patforms.

`on_install("linux", "windows", function`

**Comment by waruqi** [comment]
Does it work on your local host? Did you test it?

**Comment by BinhaoQin** [comment]
Yes, I tested using the script and it works.

**Comment by waruqi** [comment]
ok, please limit the supported patforms.

**Pull request merged**